Source/LinqToDB/Mapping/VersionBehavior.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace LinqToDB.Mapping
+{
+	/// <summary>
+	/// Defines optimistic lock column value generation strategy on record update.
+	/// Used with <see cref="OptimisticLockPropertyAttribute" /> and <see cref="ConcurrencyExtensions" /> extensions:
+	/// <list type="bullet">
+	/// <item><see cref="ConcurrencyExtensions.UpdateOptimistic{T}(IDataContext, T)"/></item>
+	/// <item><see cref="ConcurrencyExtensions.UpdateOptimisticAsync{T}(IDataContext, T, System.Threading.CancellationToken)"/></item>
+	/// <item><see cref="ConcurrencyExtensions.DeleteOptimistic{T}(IDataContext, T)"/></item>
+	/// <item><see cref="ConcurrencyExtensions.DeleteOptimisticAsync{T}(IDataContext, T, System.Threading.CancellationToken)"/></item>
+	/// </list>
+	/// </summary>
+	public enum VersionBehavior
+	{
+		/// <summary>
+		/// Column value generated by database automatically on update. E.g. using SQL Server rowversion/timestamp column or database trigger.
+		/// </summary>
+		Auto,
+		/// <summary>
+		/// Column value should be incremented by 1.
+		/// </summary>
+		AutoIncrement,
+		/// <summary>
+		/// Uses <see cref="Guid.NewGuid()"/> value.
+		/// Supported column types:
+		/// <list type="bullet">
+		/// <item><see cref="System.Guid"/></item>
+		/// <item><see cref="string"/> using <see cref="Guid.ToString()"/></item>
+		/// <item><see cref="T:byte[]"/> using <see cref="Guid.ToByteArray()"/></item>
+		/// </list>
+		/// </summary>
+		Guid
```

Comment:
prevent future changed line when adding more fields.
```suggestion
		Guid,
```